pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,226 @@
 {
+  lib,
+  stdenv,
   bzip2,
   callPackage,
-  expat,
   fetchFromGitHub,
   fontconfig,
   freetype,
   glib,
   glslang,
   harfbuzz,
-  lib,
-  libadwaita,
   libGL,
-  libpng,
   libX11,
-  libXcursor,
-  libXi,
-  libXrandr,
+  libadwaita,
   ncurses,
   nixosTests,
   oniguruma,
   pandoc,
   pkg-config,
   removeReferencesTo,
-  stdenv,
   versionCheckHook,
   wrapGAppsHook4,
   zig_0_13,
-  zlib,
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L72-L76
+  appRuntime ?
+    {
+      linux = "gtk";
+      windows = "glfw";
+    }
+    .${stdenv.hostPlatform.parsed.kernel.name} or "none",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L94
+  fontBackend ? if stdenv.hostPlatform.isDarwin then "coretext" else "fontconfig_freetype",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/PACKAGING.md#build-options
+  optimizeLevel ? "ReleaseFast",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/renderer.zig#L51-L52
+  renderer ? if stdenv.hostPlatform.isDarwin then "metal" else "opengl",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/build.zig#L106
+  withAdwaita ? true,
 }:
+
+# https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L59-L83
+assert lib.assertOneOf "appRuntime" appRuntime [
+  "glfw"
+  "gtk"
+  "none"
```

Comment:
Do we support all of these? It only makes sense to list ones we can actually build here, no matter what else upstream supports.

Reply:
GTK is the default on Linux already, and "none" is the default on MacOS (where we don't really need to do anything)

GLFW currently builds fine, but it errors on init. I don't see much harm in enumerating it here though, as hopefully we can get it to work eventually

Reply:
To reiterate: GLFW is ***never*** intended to be used in production, ever. It will not be anything close to either GTK/macOS and no work would be done to make it so. Therefore, I don't think it should be exposed to users.